### PR TITLE
[ENH] Add parameters and similarity measures to tSNE

### PR DESCRIPTION
**Pull request by pavlin-policar**
##### Issue
The Manifold Learning widget did provide tSNE, yet lacked all the important parameters with which to tune the embedding. Most notably, I find `learning_rate` to be the most important parameter and has a huge effect on the result.

##### Description of changes
Add relevant parameters (perplexity, early exaggeration, learning rate, max iterations and initialization).

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by pavlin-policar** [comment]
@lanzagar I've added the missing parameters to tSNE. This can probably be merged easily. 

I decided to leave out the Orange metrics because that would require many more changes e.g. setting precomputed parameters, computing PCA manually on the widget, all of which is doable, but should go into a separate PR.

File: Orange/widgets/unsupervised/owmanifoldlearning.py [hunk 1]
```diff
@@ -93,11 +93,29 @@ class TSNEParametersEditor(ManifoldParametersEditor):
     # rename l2 to Euclidean
     metric_values = [("l2", "Euclidean")] + metric_values
 
+    perplexity = Setting(30)
+    early_exaggeration = Setting(4)
+    learning_rate = Setting(1000)
+    n_iter = Setting(1000)
+
+    init_index = Setting(0)
+    init_values = [("random", "Random"), ("pca", "PCA")]
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owmanifoldlearning.py]: Can we leave the default init to be pca.
I would also switch their positions to have pca as the first (default) option and then random.
This would then also make it consistent with mds (pca = first & default)

File: Orange/widgets/unsupervised/owmanifoldlearning.py [hunk 2]
```diff
@@ -93,11 +93,29 @@ class TSNEParametersEditor(ManifoldParametersEditor):
...
             "metric", "Metric:")
-        self.parameters["init"] = "pca"
+        self.perplexity_spin = self._create_spin_parameter(
+            "perplexity", 0, 1000, "Perplexity:")
+        self.early_exaggeration_spin = self._create_spin_parameter(
+            "early_exaggeration", 1, 1000, "Early exaggeration:")
+        self.lr_spin = self._create_spin_parameter(
+            "learning_rate", 1, 1000, "Learning rate:")
+        self.n_iter_spin = self._create_spin_parameter(
+            "n_iter", 0, 1e5, "Max iterations:")
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owmanifoldlearning.py]: Max iterations spin allows values as low as 0. But running it with <250 results in an error. Change min value of spin control to 250.

File: Orange/widgets/unsupervised/owmanifoldlearning.py [hunk 3]
```diff
@@ -93,11 +93,29 @@ class TSNEParametersEditor(ManifoldParametersEditor):
...
+    init_values = [("random", "Random"), ("pca", "PCA")]
+
     def __init__(self, parent):
         super().__init__(parent)
+
         self.metric_combo = self._create_combo_parameter(
             "metric", "Metric:")
-        self.parameters["init"] = "pca"
+        self.perplexity_spin = self._create_spin_parameter(
+            "perplexity", 0, 1000, "Perplexity:")
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owmanifoldlearning.py]: I am not sure about this one, but sklearn doc says to consider values 5-50. I think spin limits of 1-100 should be fine?

File: Orange/widgets/unsupervised/owmanifoldlearning.py [hunk 4]
```diff
@@ -93,11 +93,29 @@ class TSNEParametersEditor(ManifoldParametersEditor):
...
+        self.early_exaggeration_spin = self._create_spin_parameter(
+            "early_exaggeration", 1, 1000, "Early exaggeration:")
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owmanifoldlearning.py]: Not sure about this one either, but 1000 sounds really high. Probably 1-100 should be fine?

File: Orange/widgets/unsupervised/owmanifoldlearning.py [hunk 5]
```diff
@@ -93,11 +93,29 @@ class TSNEParametersEditor(ManifoldParametersEditor):
     # rename l2 to Euclidean
     metric_values = [("l2", "Euclidean")] + metric_values
 
+    perplexity = Setting(30)
+    early_exaggeration = Setting(4)
+    learning_rate = Setting(1000)
+    n_iter = Setting(1000)
```
Comment (lanzagar) [inline Orange/widgets/unsupervised/owmanifoldlearning.py]: I suggest we have the same defaults as the latest sklearn version (they have been changing them recently)
perplexity=30, e.e.=12, lr=200, iter=1000

File: Orange/widgets/unsupervised/owmanifoldlearning.py (hunk 1 above)
Comment (lanzagar) [reply Orange/widgets/unsupervised/owmanifoldlearning.py]: I know this goes against the sklearn defaults, but we already had pca as the only option before and I think it makes more sense for users (random can show a completely different figure each time). The consistency with mds is an added bonus.

File: Orange/widgets/unsupervised/owmanifoldlearning.py (hunk 5 above)
Comment (pavlin-policar) [reply Orange/widgets/unsupervised/owmanifoldlearning.py]: I didn't realize they had changed them. Fixed now.

File: Orange/widgets/unsupervised/owmanifoldlearning.py (hunk 1 above)
Comment (pavlin-policar) [reply Orange/widgets/unsupervised/owmanifoldlearning.py]: As far as I can tell, both the MDS widget and manifold MDS control pane have the option to initialize randomly. I have, however, made PCA the default value, since that is what MDS does in both cases. Is there a third MDS widget that I missed?

Comment (lanzagar) [reply Orange/widgets/unsupervised/owmanifoldlearning.py]: This is fine now. That is what I meant - just change the order of pca and random, so that pca is default (but we still have both options, like in mds)
